### DISCUSSION TITLE
Add minimum off time setting to thermostat

### DISCUSSION THREAD

**Discussion by jsiddall** [arendst/Tasmota#17940] [Ideas]
The current thermostat implementation allows a minimum run time (TimeMinActionSet) but not a minimum off time.  Heat pump/air conditioning systems with compressors often need a few minutes off time after running for pressure in the system to equalize before the compressor can be restarted.  This setting would enforce that minimum off time.

The minimum off time setting, which could default to 0 seconds to preserve the current behavior, should apply regardless of mode or any other settings changes.  It should also apply at bootup so that, for example, a brief power outage does not cause immediate the re-activation of the output when the Tasmota starts.

**Comment by Wintespe** [comment]
If you like, use the Tasmota PID controller. I've been using the controller for some time with great success.

The following commands are useful for your application:

**PidAuto**
Auto mode 1 or 0. This can be used to enable or disable the control (1=enable, auto mode, 0=disabled, manual mode). When in manual mode the output is set to the value defined for PidManualPower

**PidManualPowe**r
Power output when in manual or fallback mode.

### @ your problem

Set PidManualPower to 0

Write a **rule** with the following implementation:

In the **Power OFF** trigger of your output, set PidAuto to 0 (manual mode) and start a timer for xx seconds

If timer fires set PidAuto to 1 (auto mode)

I haven't tried it, but I'm pretty sure it works. I was able to verify PidAuto in my application and it is ok.

**Comment by jsiddall** [comment]
I guess a rule could be tied into the thermostat to make the holdoff on power-on work.  But there are likely other use cases where the relay could go off and on again in a short time.  For example, if the setpoint was set high then low again (or vice versa for heating).  Thinking a more generalized solution would be good.
